numpy/_core/src/multiarray/stringdtype/casts.cpp

```diff
@@ -2072,70 +2240,90 @@ get_casts()
     casts[cast_i++] = StringToUnicodeCastSpec;
     casts[cast_i++] = StringToBoolCastSpec;
     casts[cast_i++] = BoolToStringCastSpec;
-    casts[cast_i++] = StringToInt8CastSpec;
-    casts[cast_i++] = Int8ToStringCastSpec;
-    casts[cast_i++] = StringToInt16CastSpec;
-    casts[cast_i++] = Int16ToStringCastSpec;
-    casts[cast_i++] = StringToInt32CastSpec;
-    casts[cast_i++] = Int32ToStringCastSpec;
-    casts[cast_i++] = StringToInt64CastSpec;
-    casts[cast_i++] = Int64ToStringCastSpec;
-    casts[cast_i++] = StringToUInt8CastSpec;
-    casts[cast_i++] = UInt8ToStringCastSpec;
-    casts[cast_i++] = StringToUInt16CastSpec;
-    casts[cast_i++] = UInt16ToStringCastSpec;
-    casts[cast_i++] = StringToUInt32CastSpec;
-    casts[cast_i++] = UInt32ToStringCastSpec;
-    casts[cast_i++] = StringToUInt64CastSpec;
-    casts[cast_i++] = UInt64ToStringCastSpec;
+
+    casts[cast_i++] = getStringToIntCastSpec<npy_int8,  npy_longlong, NPY_INT8>();
```

Comment:
I suppose we don't have it yet, although some of Sayed's work probably includes it.  We really should have a `typenum_to_type<NPY_INT8>()` and `type_to_typenum<npy_int8>()` to get back and forth these and avoid the duplication for int8.